emperor/support_files/js/view.js

```diff
@@ -101,37 +103,55 @@ DecompositionView.prototype._initBaseView = function() {
 
   hasConfidenceIntervals = this.decomp.hasConfidenceIntervals();
 
-  this.decomp.apply(function(plottable) {
-    mesh = new THREE.Mesh(geometry, new THREE.MeshPhongMaterial());
-    mesh.name = plottable.name;
+  if (this.decomp.isScatterType()) {
```

Comment:
there is quite a bit of work going on within this if-statement, would it make sense to decompose?

Reply:
Yes, I also noticed this and even worse there's some duplicated code with this and other method. I've refactored this in #647